### ISSUE TITLE
Color Replacement Brush Not Working Correctly?

### ISSUE THREAD

**Issue by camoody1** [photopea/photopea#5876]
I've not used this brush before tonight, but someone asked for help using it on Reddit, and after playing around with it in both Photopea and Photoshop, it seems to be working incorrectly in Photopea.

Here is the sample video I was following:
https://www.youtube.com/watch?v=DJq7B2WYh8M&ab_channel=ARTOSE

I took a screen shot of the three bell peppers she has stacked on top of each other at the 2:13 mark in the video so that I would be using the same image as her. I put the image into Photoshop 2023, chose a Blue color, set the Color Replacement brush to Hue, select color once and a tolerance of 33%. When I drew over the green and yellow peppers, they turned a blue color very close to what I had selected for my color swatch. But when I tried the same image and settings in Photopea, there was almost no noticeable difference in the color on the image. If anything, there was just as slight blue haze painted across the peppers. Unless I'm missing something else, I don't think Photopea is handling this tool correctly.

Image I used:
![image](https://github.com/photopea/photopea/assets/73725653/b1150caf-11c3-4786-8f19-ee1a5e47154c)

Photoshop Result:
![image](https://github.com/photopea/photopea/assets/73725653/62681175-de67-4f8d-9526-c4e926492544)

Photopea Result:
![image](https://github.com/photopea/photopea/assets/73725653/1382edfc-e8e5-4995-a137-5450afa7cb22)

**Comment by photopea** [comment]
I just tried it and the pepper turns blue for me in Photopea :(  could you attach a screenshot of the whole editor after painting with blue in Photopea?

**Comment by camoody1** [comment]
Okay, okay... I just figured something out. I got a new mouse a couple of days ago with, like 12 buttons on the side. I was testing this tool, again, and everything was working totally correctly. Blue was nice and dark like it's supposed to be. Then, I accidently hit one of the buttons on the side of my mouse, and it all went super pale again. I could barely see the blue. Apparently, on the Windows desktop, these side buttons equate to keystrokes and the button I accidently pressed was the number "1". So, I have no idea what the number "1" changes in Photopea, but when I pressed "0", it started working correctly, again. So, my question now is... what the heck do the number keys change in Photopea? It's not opacity of the brush and I don't notice anything else changing. But clearly, something is very different.

**Comment by camoody1** [comment]
Okay, follow-up...

I switched to the Brush tool and started pressing the number keys. I see now that those keys DO change the opacity of the brush. But the numbers and opacity do not always match up. You would think that 1 = 10%, 2 = 20%, 3 = 30%..... 0 = 100%. But it doesn't always work that way. Sometimes, the opacity only goes up by 1% or 4%. Sometimes, pressing the same number twice results in the opacity changing.

So, while I've definitely figured out my problem with the Color Replacement tool, I think I've found a new issue with the number keys and how they scale Brush Opacity.

Also, could you maybe add an Opacity box to the toolbar for the Color Replacement tool since Brush Opacity actually DOES have a marked effect on its results?

**Comment by photopea** [comment]
Yeah, the opacity input is removed from the top menu, but I guess it can still be changed by the keyboard shortcuts.

BTW. switch to the Brush tool to see what it does. It is like "typing" the value.

**Comment by camoody1** [comment]
> Yeah, the opacity input is removed from the top menu, but I guess it can still be changed by the keyboard shortcuts.
> 
> BTW. switch to the Brush tool to see what it does. It is like "typing" the value.

Yeah, that's how I noticed what was happening. I switched to the Brush tool and noticed the Opacity was changing. I see now that if you wait long enough between key presses, the Opacity goes up in 10% increments. But if you press buttons quickly one after another, it's like you're just typing the Opacity. I assume that is intentional? I guess I will still think that the Opacity field should be shown for the Color Replacement tool since the tool is definitely affected by it.
